staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package encryptionconfig
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/storage/value"
+	"k8s.io/apiserver/pkg/storage/value/encrypt/identity"
+)
+
+// TransformerConfig is a blanket interface for all provider configurations
+type TransformerConfig interface {
+	// GetPrefixTransformer constructs and returns a PrefixTransformer using the provided configuration
+	GetPrefixTransformer() (value.PrefixTransformer, error)
+	// IsNil abstracts the nil check for the interface, since nil pointer checks on interface don't work
+	IsNil() bool
+}
+
+// ProviderConfig stores the provided configuration for an encryption provider
+type ProviderConfig struct {
+	AES      *AESConfig      `json:"aes,omitempty"`
+	Identity *IdentityConfig `json:"identity,omitempty"`
+}
+
+// GetPrefixTransformer returns the PrefixTransformer from the provider configuration,
+// and returns an error if more than one providers were specified
+func (config *ProviderConfig) GetPrefixTransformer() (value.PrefixTransformer, error) {
```

Comment:
We don't put methods on API objects.  These should be separate from the config.  Everything that is a struct that represents config should be in a single file types.go (see pkg/apis/componentconfig for an example), and anything that reads config should take those as inputs.  You should generate a Transformer from the top level config, but this is all stuff that has nothing to do with the objects.